.github/workflows/danger.yml

```diff
@@ -5,12 +5,23 @@ on:
     # when the labels change, not only when a PR is opened/reopened or changes
     # are pushed to it.
     types: [opened, reopened, synchronize, labeled, unlabeled]
+  # In order to validate the milestone, we need to listen for the event on the
+  # issue, as there's no event for milestones in PRs –GitHub PRs are a special
+  # kind of issue. Un
+  issues:
+    types: [milestoned, demilestoned]
```

Comment:
It seems the triggers for milestones don't run correctly:

- If I open a new PR with no milestone and no labels, I see warnings about both
- If I then add a milestone, the warning doesn't go away
- If I then add a label, both warnings go away

It turns out this is an issue with Peril as well – it has the same bug. Is it potentially related to the `if` condition below?

Reply:
I understood why that happened in #14336.